### PR TITLE
 Replace OpenTracing error logs with SignalFx tags

### PR DESCRIPTION
**Pull request by owais**
This PR changes how errors are recorded on spans. It uses the new
SignalFx semantic convention and records the errors as attributes
instead of logs.

------------

#### OpenTracing Patch

I've tried to patch OpenTracing Span's `_on_error` method to record errors as tags instead of logs. This allows us to continue using python-opentracing without forking it.

### CODE REVIEW

File: tests/unit/conftest.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            assert k in span.tags
+            assert span.tags[k] == v
+
+    def assert_span_with_exception(self, span, exc, min_tb_length=50):
+        tags = span.tags
+        assert tags[ext_tags.ERROR] is True
+        assert tags[ext_tags.ERROR_MESSAGE] == str(exc)
+        assert tags[ext_tags.ERROR_OBJECT] == str(exc.__class__)
+        assert tags[ext_tags.ERROR_KIND] == exc.__class__.__name__
+        tb = reduce(lambda i, j: i + j, tags[ext_tags.ERROR_STACK])
```
Comment (rmfitzpatrick) [inline tests/unit/conftest.py:19]: I think `''.join(tags[ext_tags.ERROR_STACK])` is clearer, but why is this necessary?

Comment (owais) [reply tests/unit/conftest.py:19]: Ah, of course. Been away from python for too long I guess :)

File: signalfx_tracing/patch_span.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        return
+
+    span.set_tag(tags.ERROR, True)
+    span.set_tag(tags.ERROR_MESSAGE, str(exc_val))
+    span.set_tag(tags.ERROR_OBJECT, str(exc_val.__class__))
+    span.set_tag(tags.ERROR_KIND, exc_type.__name__)
+    span.set_tag(tags.ERROR_STACK, traceback.format_tb(exc_tb))
+
+
+opentracing.Span._on_error = _patched_on_error
```
Comment (rmfitzpatrick) [inline signalfx_tracing/patch_span.py:20]: I don't believe this was always a staticmethod (2.1), so using wrapt's `wrap_function_wrapper('opentracing', 'Span._on_error', _patched_on_error)` will handle both cases w/ proper descriptor protocol usage.

File: tests/unit/conftest.py (hunk 1 above)
Comment (owais) [reply tests/unit/conftest.py:19]: > but why is this necessary?

MockTracer uses an internal MockSpan which has it's own implementation of `set_tag` which sets python objects as tag values directly instead of their string representations. So this turns out as a list instead of a string in tests.
